public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -529,557 +536,680 @@ public SurveyQuestionResult apply(AchillesResult o) {
-                    if (sourceParent != null) {
-                        standardParent.setSourceCount(sourceParent.getCount());
+        try {
+            CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+            List<CBCriteria> criteriaList = criteriaDao.findParentCounts(String.valueOf(conceptId), domainId.toUpperCase(), new String(domainId+"_rank1"));
+            Multimap<String, CBCriteria> criteriaRowsByConcept = Multimaps.index(criteriaList, CBCriteria::getConceptId);
+            CriteriaParentResponse response = new CriteriaParentResponse();
+            if (criteriaList.size() > 0) {
+                List<CBCriteria> parentList = criteriaRowsByConcept.get(String.valueOf(conceptId)).stream().collect(Collectors.toList());
+                CBCriteria parent = null;
+                CBCriteria standardParent = null;
+                CBCriteria sourceParent = null;
+                if (parentList.size() > 1) {
+                    List<CBCriteria> standardParentList = parentList.stream().filter(p -> p.getStandard() == true).collect(Collectors.toList());
+                    standardParent = (standardParentList != null && standardParentList.size() > 0) ? standardParentList.get(0) : null;
+                    List<CBCriteria> sourceParentList = parentList.stream().filter(p -> p.getStandard() == false).collect(Collectors.toList());
+                    sourceParent = (sourceParentList != null && sourceParentList.size() > 0) ? sourceParentList.get(0) : null;
+                    if (standardParent != null) {
+                        if (sourceParent != null) {
+                            standardParent.setSourceCount(sourceParent.getCount());
+                        }
+                        parent = standardParent;
+                    } else {
+                        parent = sourceParent;
                     }
-                    parent = standardParent;
                 } else {
-                    parent = sourceParent;
+                    parent = parentList.get(0);
                 }
-            } else {
-                parent = parentList.get(0);
-            }
-            if (criteriaList.size() >= 1) {
-                criteriaList.remove(parent);
+                if (criteriaList.size() >= 1) {
+                    criteriaList.remove(parent);
+                }
+                response.setParent(TO_CLIENT_CBCRITERIA.apply(parent));
+                Multimap<Long, CBCriteria> parentCriteria = Multimaps
+                        .index(criteriaList, CBCriteria::getParentId);
+                return ResponseEntity.ok(response);
             }
-            response.setParent(TO_CLIENT_CBCRITERIA.apply(parent));
-            Multimap<Long, CBCriteria> parentCriteria = Multimaps
-                    .index(criteriaList, CBCriteria::getParentId);
             return ResponseEntity.ok(response);
+        } catch(Exception ie) {
+            if (ExceptionUtils.isSocketTimeoutException(ie)) {
```

Comment:
Why are we catching a SocketTimeoutException and throwing a ServerErrorException?